sklearn/cluster/_birch.py

```diff
@@ -586,8 +577,8 @@ def predict(self, X):
         labels : ndarray of shape(n_samples,)
             Labelled data.
         """
-        X = check_array(X, accept_sparse='csr')
-        self._check_fit(X)
+        check_is_fitted(self)
+        X = self._validate_data(X, accept_sparse='csr', reset=False)
```

Comment:
just as a note, we don't call `_validate_data` in predict yet so this one would be an outlier.

It's up to you whether you want to do this here or in the other PR since you're in charge of this now :p

Reply:
I reverted this. I will introduce `_validate_data` in `predict` in another PR.